examples/styled-jsx-with-csp/pages/_document.jsx

```diff
@@ -1,17 +1,17 @@
 import Document, { Html, Head, Main, NextScript } from 'next/document'
-import flush from 'styled-jsx/server'
-
 import { nanoid } from 'nanoid'
+import { StyleRegistry, createStyleRegistry } from 'styled-jsx'
 
 class CustomDocument extends Document {
   static async getInitialProps(ctx) {
     const nonce = nanoid()
+    const jsxStyleRegistry = createStyleRegistry()
 
     // https://github.com/vercel/next.js/blob/canary/packages/next/pages/_document.tsx#L89
     const { html, head } = await ctx.renderPage()
```

Comment:
Aren't the styles being lost from rendering the page since we aren't calling `Document.getInitialProps` which collects the styles? Also should `StyleRegistry` be in `_document` at all since `<style jsx>` tags shouldn't be rendered in here. Seems like we should be adding the `nonce` to the children in `styles` returned from `Document.getInitialProps`

x-ref: https://github.com/vercel/next.js/blob/b8a552b93b80b0a2acd9c08e9ba7ab61553b0357/test/development/basic/styled-components/pages/_document.js#L16
x-ref: https://github.com/vercel/next.js/blob/b8a552b93b80b0a2acd9c08e9ba7ab61553b0357/packages/next/server/render.tsx#L647

Reply:
Added 2nd arg `option` for `defaultGetInitialProps` that can make the customization simpler, and avoid the multiple nested `StyleRegistry`.